src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/TopicPartitionWriter.java

```diff
@@ -1,270 +1,272 @@
-                            exception);
-                }
-            });
-        } catch (IllegalStateException e) {
-            log.error("Failed to write records to miscellaneous dead-letter queue topic, "
-                    + "kafka producer has already been closed. Exception={0}", e);
-        }
-    }
-
-    String getFilePath(@Nullable Long offset) {
-        // Should be null if flushed by interval
-        offset = offset == null ? currentOffset : offset;
-        long nextOffset = fileWriter != null && fileWriter.isDirty() ? offset + 1 : offset;
-
-        return Paths.get(basePath, String.format("kafka_%s_%s_%d.%s%s", tp.topic(), tp.partition(), nextOffset,
-                ingestionProps.ingestionProperties.getDataFormat(), COMPRESSION_EXTENSION)).toString();
-    }
-
-    void writeRecord(SinkRecord record) throws ConnectException {
-        if (record != null) {
-            try (AutoCloseableLock ignored = new AutoCloseableLock(reentrantReadWriteLock.readLock())) {
-                this.currentOffset = record.kafkaOffset();
-                fileWriter.writeData(record);
-            } catch (IOException | DataException ex) {
-                handleErrors(record, ex);
-            }
-        }
-    }
-
-    private void handleErrors(SinkRecord record, Exception ex) {
-        if (BehaviorOnError.FAIL == behaviorOnError) {
-            throw new ConnectException(FILE_EXCEPTION_MESSAGE, ex);
-        } else if (BehaviorOnError.LOG == behaviorOnError) {
-            log.error(FILE_EXCEPTION_MESSAGE, ex);
-            sendFailedRecordToDlq(record);
-        } else {
-            log.debug(FILE_EXCEPTION_MESSAGE, ex);
-            sendFailedRecordToDlq(record);
-        }
-    }
-
-    void open() {
-        // Should compress binary files
-        fileWriter = new FileWriter(
-                basePath,
-                fileThreshold,
-                this::handleRollFile,
-                this::getFilePath,
-                flushInterval,
-                reentrantReadWriteLock,
-                ingestionProps.ingestionProperties.getDataFormat(),
-                behaviorOnError,
-                isDlqEnabled);
-    }
-
-    void close() {
-        try {
-            fileWriter.rollback();
-            fileWriter.close();
-        } catch (IOException e) {
-            log.error("Failed to rollback with exception={0}", e);
-        }
-        try {
-            if (dlqProducer != null) {
-                dlqProducer.close();
-            }
-        } catch (Exception e) {
-            log.error("Failed to close kafka producer={0}", e);
-        }
-        try {
-            FileUtils.deleteDirectory(new File(basePath));
-        } catch (IOException e) {
-            log.error("Unable to delete temporary connector folder {}", basePath);
-        }
-    }
-
-    void stop() {
-        fileWriter.stop();
-    }
-}
+package com.microsoft.azure.kusto.kafka.connect.sink;
```

Comment:
This file had CRLF line endings, which I consider to be worse, and is inconsistent with the rest of our code.